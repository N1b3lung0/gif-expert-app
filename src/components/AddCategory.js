import React, { useState } from 'react'
import PropTypes from 'prop-types';
import { InputText } from 'primereact/inputtext';


export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => setInputValue( e.target.value );

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories( categories => [inputValue, ...categories] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <span className="p-float-label">
                <InputText
                    id="float-input"
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange }
                />
                <label htmlFor="float-input">Search</label>
            </span>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}