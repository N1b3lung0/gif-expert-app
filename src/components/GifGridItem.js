import 'primereact/resources/themes/nova-dark/theme.css';
import 'primereact/resources/primereact.css';

import React from 'react'
import PropTypes from 'prop-types';
import { Card } from 'primereact/card';

export const GifGridItem = ({ title, url}) => {
    
    const header = ( <img src={ url } alt={ title }/> );

    return (
        <Card
            subTitle={ title }
            style={{width: '360px'}}
            className="ui-card-shadow card animate__animated animate__fadeIn"
            header={ header }
        />
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

Card.propTypes = {
    subTitle: PropTypes.string.isRequired,
    header: PropTypes.object.isRequired
}
