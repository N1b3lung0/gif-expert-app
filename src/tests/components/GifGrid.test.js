import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Testing GifGrid component', () => {

    const category = 'Friends';
    
    test('should show the component GifGrid correctly', () => {

        useFetchGifs.mockReturnValue({
            images: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('should show items when useFetchGifs load images', () => {

        const gifs = [{
            id: '123',
            url: 'https://localhost/imagen.jpg',
            title: 'title'
        }];

        useFetchGifs.mockReturnValue({
            images: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
    
})
