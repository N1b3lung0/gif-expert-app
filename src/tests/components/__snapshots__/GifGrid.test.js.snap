// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component should show items when useFetchGifs load images 1`] = `
<Fragment>
  <h3>
    Friends
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123"
      key="123"
      title="title"
      url="https://localhost/imagen.jpg"
    />
  </div>
</Fragment>
`;

exports[`Testing GifGrid component should show the component GifGrid correctly 1`] = `
<Fragment>
  <h3>
    Friends
  </h3>
  <p>
    Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
